skxray/core/xsvs.py

```diff
@@ -0,0 +1,289 @@
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+"""
+    X-ray speckle visibility spectroscopy(XSVS) - Dynamic information of
+    the speckle patterns  are obtained by analyzing the photon statistics
+    and calculating the speckle contrast in single scattering patterns.
+
+    This module will provide XSVS analysis tools
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+import numpy as np
+import time
+
+import skxray.core.correlation as corr
+import skxray.core.roi as roi
+from skxray.core.utils import bin_edges_to_centers, geometric_series
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def xsvs(image_sets, label_array, timebin_num=2, number_of_img=50,
+         max_cts=None):
+    """
+    This function will provide the probability density of detecting photons
+    for different integration time.
+
+    The experimental probability density P(K) of detecting photons K is
+    obtained by histogramming the photon counts over an ensemble of
+    equivalent pixels and over a number of speckle patterns recorded
+    with the same integration time T under the same condition.
+
+    Parameters
+    ----------
+    image_sets : array
+        sets of images
+    label_array : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+    timebin_num : int, optional
+        integration times
+    number_of_img : int, optional
+        number of images
+    max_cts : int, optional
+       the brightest pixel in any ROI in any image in the image set.
+
+    Returns
+    -------
+    prob_k_all : array
+        probability density of detecting photons
+    prob_k_std_dev : array
+        standard deviation of probability density of detecting photons
+
+    Note
+    ----
+    These implementation is based on following references
+    References: text [1]_, text [2]_
+
+    .. [1] L. Li, P. Kwasniewski, D. Oris, L Wiegart, L. Cristofolini,
+       C. Carona and A. Fluerasu , "Photon statistics and speckle visibility
+       spectroscopy with partially coherent x-rays" J. Synchrotron Rad.,
+       vol 21, p 1288-1295, 2014.
+
+    .. [2] R. Bandyopadhyay, A. S. Gittings, S. S. Suh, P.K. Dixon and
+       D.J. Durian "Speckle-visibilty Spectroscopy: A tool to study
+       time-varying dynamics" Rev. Sci. Instrum. vol 76, p  093110, 2005.
+
+    """
+    if max_cts is None:
+        max_cts = roi.roi_max_counts(image_sets, label_array)
+
+    # number of ROI's
+    num_roi = np.max(label_array)
+
+    # create integration times
+    time_bin = geometric_series(timebin_num, number_of_img)
```

Comment:
Does it make more sense for the user to provide the list of integration times that they want to use as an input parameter instead of forcing them to use a geometric series?